### PR TITLE
Week 5 vanessa bastien

### PR DESCRIPTION
**Pull request by VKathryn**
# Submitting Week 5 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [ ] JavaScript Coding Challenges
- [ ] Ruby Coding Challenges
- [X ] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here: Obviously, I got only the interview questions done for this assessment this week.

### CODE REVIEW

**Comment by WalkerTexas-coder** [comment]
Anything I can do to help out with getting the assessment in on time?

File: interview-questions.md [hunk 1]
```diff
@@ -10,42 +10,46 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 Your answer:
 
-Researched answer:
+Researched answer: Hashes in Ruby consist of key:value pairs. It is similar to an array, yet different because the indexing is not static as it is with arrays. In hashes, indexing is done with arbitrary keys of an object type. You create a hash by using the arrow function to assign a key to a value key => value and separating each key value pair with a comma. The entire has should be enclosed in curly braces. 
+
```
Comment (SunkissedQueen) [inline interview-questions.md:14]: would suggest including the relationship of the datatype of the key and the use of the hash rocket

File: interview-questions.md [hunk 2]
```diff
@@ -10,42 +10,46 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 4. What is a relational database? Are there other kinds of databases?
 
-Your answer:
+Your answer: A database is an organized collection of data. A relational database is a database in which we understand how data relates to other data. 
+
+Researched answer: A relational database is a model created in the 1970's by E.F. Codd from IBM. It allows tables to be related to one another using a common attribute. In this way we can understand how data is related to other data and use that knowledge to call on it effectively. For example, a library with an online lending system would have a table that contained data for the library user, such as Library ID, name, address, and preferred library location. The system would also have a table containing requested items which would include data such as Media ID, Library ID, current location of media, and Date of Delivery to preferred location. These tables are now related due to the shared Library ID, which enables you to utilize a foreign key to connect the tables. Some examples of more complicated relational databases are Cloud SQL and AlloyDB.
```
Comment (SunkissedQueen) [inline interview-questions.md:33]: good examples. I would say add more clarity on how that foreign key would appear also include the model association

File: interview-questions.md [hunk 3]
```diff
@@ -10,42 +10,46 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-Researched answer:
+There are certainly more kinds of databases! One is a centralized database, which is when the data is stored in one central location and users access the data from different locations. There are also the following databases: personal, commercial, end user, operational, cloud, object-oriented, graph, noSQL, and more beyond that I'm sure!
 
 5. What are primary keys? Why are they important?
 
-Your answer:
+Your answer: Primary keys are a crucial reference point when working with Ruby on Rails. Similarly to how you use an index to pinpoint locations of data in JavaScript, you use primary keys to indicate specific data. 
 
-Researched answer:
+Researched answer: The primary key contains values (never null values) and identifies the rows in a table. It is a relational database table column which is meant to uniquely identify each table value. It is used to ensure that the data in a specific column is unique.
```
Comment (SunkissedQueen) [inline interview-questions.md:41]: examples would be great content

**Review by SunkissedQueen — APPROVED** [review]
Vanessa, you might have some conflicts since there are changes to the same file on the other PR.

**Pull request merged**